packages/jupyter-ai/jupyter_ai/document_loaders/directory.py

```diff
@@ -126,16 +132,24 @@ def split(path, all_files: bool, splitter):
                 ]
                 filenames = [f for f in filenames if not f[0] == "."]
             filepaths += [Path(os.path.join(dir, filename)) for filename in filenames]
+        filepaths = [
+            fp
+            for fp in filepaths
+            if fp.suffix.lower() in {j.lower() for j in SUPPORTED_EXTS}
```

Comment:
This line will be creating a set of supported extensions `{j.lower() for j in SUPPORTED_EXTS}` for every filepath. Given that supported extensions would not change during the function runtime, create supported extensions set somewhere in the function and reuse it when needed:
```py
supported_extensions = {j.lower() for j in SUPPORTED_EXTS}
...
fp.suffix.lower() in supported_extensions
```

Then instead of O(n*m) where n is the number of files and m is the number of supported extensions the complexity of this line would be reduced to O(n + m).